addons/auth_signup/controllers/main.py

```diff
@@ -50,11 +50,15 @@ def web_auth_signup(self, *args, **kw):
             raise werkzeug.exceptions.NotFound()
 
         if 'error' not in qcontext and request.httprequest.method == 'POST':
-            try:
-                self.do_signup(qcontext)
-                return super(AuthSignupHome, self).web_login(*args, **kw)
-            except (SignupError, AssertionError), e:
-                qcontext['error'] = _(e.message)
+            if request.env["res.users"].sudo().search(
```

Comment:
I thought about it, but since there is no `DuplicatedLoginError` exception, making the code fail and then searching inside the string of the exception did not seem much better than just avoiding the exception before reaching it.

And if I search for an existing user after calling `do_signup()`, there is no way to know if the exception was raised before or after creating a record, so doing a login search after the exception could lead to false positives.